restapi/src/main/java/io/stargate/web/docsapi/service/query/search/resolver/impl/OrExpressionDocumentsResolver.java

```diff
@@ -0,0 +1,180 @@
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class OrExpressionDocumentsResolver implements DocumentsResolver {
+
+  private final Or<FilterExpression> expression;
+
+  private final List<AbstractSearchQueryBuilder> queryBuilders;
+
+  private final ExecutionContext context;
+
+  public OrExpressionDocumentsResolver(Or<FilterExpression> expression, ExecutionContext context) {
+    this.expression = expression;
+    this.queryBuilders = buildQueries(getChildren(expression));
+    this.context = createContext(context, expression);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public Flowable<RawDocument> getDocuments(
+      QueryExecutor queryExecutor,
+      DocsApiConfiguration configuration,
+      String keyspace,
+      String collection,
+      Paginator paginator) {
+    DataStore dataStore = queryExecutor.getDataStore();
+
+    // find the max size of columns in all queries and use that for now
+    String[] columns =
+        queryBuilders.stream()
+            .map(qb -> columnsForQuery(qb, configuration.getMaxDepth()))
+            .max(Comparator.comparingInt(o -> o.length))
+            .orElseGet(() -> QueryConstants.ALL_COLUMNS_NAMES.apply(configuration.getMaxDepth()));
+
+    return Flowable.fromIterable(queryBuilders)
+        .flatMap(
+            queryBuilder ->
+                RxUtils.singleFromFuture(
+                        () -> {
+                          BuiltQuery<? extends BoundQuery> query =
+                              queryBuilder.buildQuery(
+                                  dataStore::queryBuilder, keyspace, collection, columns);
+                          return dataStore.prepare(query);
+                        })
+                    .toFlowable())
+        .toList()
+        .cache()
+        .flatMapPublisher(
+            preparedQueries -> {
+              // find all with empty bind values
+              List<BoundQuery> boundQueries =
+                  preparedQueries.stream().map(Query::bind).collect(Collectors.toList());
+
+              // execute them all by respecting the paging state
+              return queryExecutor.queryDocs(
+                  boundQueries,
+                  configuration.getSearchPageSize(),
+                  paginator.getCurrentDbPageState(),
+                  context);
+            })
+        .filter(
+            doc -> {
+              // now we can get doc as a result of the persistence or in memory query
+              // if we only run persistence queries we can return true immediately
+              // running only persistence queries means we had no path columns
+              // since all rows must have same columns we can find any that does not have it
+              boolean noPaths =
+                  doc.rows().stream()
+                      .anyMatch(r -> !r.columnExists(QueryConstants.P_COLUMN_NAME.apply(0)));
```

Comment:
nit: I believe we can precompute this flag outside of the Rx pipeline (for the sake of optimization)

Reply:
Good catch. Done.